starter-code/ShoppingListWithDetailView/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistdetailview/ShoppingListAdapter.java

```diff
@@ -36,17 +37,10 @@ public void onBindViewHolder(final ShoppingItemViewHolder holder, int position)
         holder.mNameTextView.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                // Get a reference to the MainActivity as a Context
-                Context mainActivity = holder.mNameTextView.getContext();
-
-                // Create the intent
-                Intent intent = new Intent(mainActivity, DetailActivity.class);
-
-                // Add the ID as an extra
-                intent.putExtra(DetailActivity.ITEM_ID_KEY, currentItem.getId());
-
-                // Start the detail activity
-                mainActivity.startActivity(intent);
+                mListener.onItemSelected(mShoppingItems.get(position).getName(),
+                        mShoppingItems.get(position).getDescription(),
+                        mShoppingItems.get(position).getPrice(),
+                        mShoppingItems.get(position).getType());
```

Comment:
This is another place where you could pass just the ID value for the selected item